tools/src/test/java/com/nedap/archie/creation/ExampleJsonInstanceGeneratorTest.java

```diff
@@ -104,45 +118,73 @@ public void parseOrdinal() throws Exception {
+    public void generateAllCKMExamples2() throws Exception {
+        ExampleJsonInstanceGenerator structureGenerator = createExampleJsonInstanceGenerator();
         FullArchetypeRepository repository = TestUtil.parseCKM();
         ObjectMapper mapper = new ObjectMapper();
         mapper.enable(SerializationFeature.INDENT_OUTPUT);
         int numberCreated = 0, validationFailed = 0, generatedException = 0, jsonSchemaValidationRan = 0, jsonSchemaValidationFailed = 0;
+        int secondJsonSchemaValidationRan = 0, reserializedJsonSchemaValidationFailed = 0;
         repository.compile(BuiltinReferenceModels.getMetaModels());
-        JsonSchemaValidator jsonSchemaValidator = new JsonSchemaValidator(BuiltinReferenceModels.getBmmRepository().getModel("openehr_rm_1.0.4").getModel());
+        BmmModel model = BuiltinReferenceModels.getBmmRepository().getModel("openehr_rm_1.0.4").getModel();
+        JsonSchemaValidator firstValidator = new JsonSchemaValidator(model, true);
+        JsonSchemaValidator secondValidator = new JsonSchemaValidator(model,false);
+
+        ObjectMapper archieObjectMapper = getArchieObjectMapper();
+
         for(ValidationResult result:repository.getAllValidationResults()) {
-            if(result.passes()) {
+            if(result.passes()
+                ) {
                 String json = "";
                 try {
                     Flattener flattener = new Flattener(repository, BuiltinReferenceModels.getMetaModels()).createOperationalTemplate(true);
                     OperationalTemplate template = (OperationalTemplate) flattener.flatten(result.getSourceArchetype());
                     Map<String, Object> example = structureGenerator.generate(template);
                     json = mapper.writeValueAsString(example);
-                    JacksonUtil.getObjectMapper().readValue(json, RMObject.class);
+
+                    RMObject parsed = archieObjectMapper.readValue(json, RMObject.class);
                     numberCreated++;
-                   // if(Sets.newHashSet("COMPOSITION", "OBSERVATION", "EVALUATION", "INSTRUCTION", "SECTION", "ACTION").contains(template.getDefinition().getRmTypeName())) {
-                        jsonSchemaValidationRan++;
-                        jsonSchemaValidator.validate(template.getDefinition().getRmTypeName(), json);
-                   // }
 
+                    jsonSchemaValidationRan++;
+                    List<Problem> problems = firstValidator.validate(json);
+                    if(problems.size() > 0) {
+                        logger.error("validation failed for {}", result.getArchetypeId());
+                        logger.error(Joiner.on("\n").join(problems));
+                        jsonSchemaValidationFailed++;
+                        continue;
+                    }
+                    logger.error("first validation ok for {}", result.getArchetypeId());
```

Comment:
This is not an error.
```suggestion
                    logger.info("first validation ok for {}", result.getArchetypeId());
```